crates/storage/provider/src/bundle_state/state_reverts.rs

```diff
@@ -78,30 +67,54 @@ impl StateReverts {
                     }
                 }
 
-                tracing::trace!(target: "provider::reverts", ?address, ?storage, "Writing storage reverts");
+                tracing::trace!(target: "provider::reverts", %address, ?storage, "Writing storage reverts");
                 for (key, value) in StorageRevertsIter::new(storage, wiped_storage) {
-                    storage_changeset_cursor.append_dup(storage_id, StorageEntry { key, value })?;
+                    if should_append_storage {
+                        storage_changeset_cursor
+                            .append_dup(storage_id, StorageEntry { key, value })?;
+                    } else {
+                        if let Some(entry) =
+                            storage_changeset_cursor.seek_by_key_subkey(storage_id, key)?
+                        {
+                            if entry.key == key {
+                                tracing::warn!(target: "provider::reverts", ?storage_id, ?entry, "Overwriting storage changeset entry");
+                                storage_changeset_cursor.delete_current()?;
+                            }
+                        }
+
+                        storage_changeset_cursor.upsert(storage_id, StorageEntry { key, value })?;
+                    }
```

Comment:
we don't do this here because `block-number||address keys` are unique

Reply:
you cannot assume this for all uses of bundle state